crates/core/tedge/src/cli/upload/mod.rs

```diff
@@ -0,0 +1,94 @@
+use crate::command::BuildCommand;
+use crate::command::BuildContext;
+use crate::command::Command;
+use crate::ConfigError;
+use c8y_api::http_proxy::C8yEndPoint;
+use std::collections::HashMap;
+use std::path::PathBuf;
+use tedge_config::ProfileName;
+
+mod c8y;
+
+#[derive(clap::Subcommand, Debug)]
+pub enum UploadCmd {
+    /// Upload a file to Cumulocity
+    ///
+    /// The command create a new event for the device,
+    /// attach the given file content to this new event,
+    /// and return the event ID.
+    C8y {
+        /// Path to the uploaded file
+        #[clap(long)]
+        file: PathBuf,
+
+        /// MIME type of the file content
+        #[clap(long, default_value = "application/octet-stream")]
+        mime_type: String,
+
+        /// Type of the event
+        #[clap(long = "type", default_value = "tedge_UploadedFile")]
+        event_type: String,
+
+        /// Text description of the event. Defaults to "Uploaded file: <FILE>"
+        #[clap(long)]
+        text: Option<String>,
+
+        /// JSON fragment attached to the event
+        #[clap(long, default_value = "{}")]
+        #[arg(value_parser = parse_json)]
+        json: HashMap<String, serde_json::Value>,
+
+        /// Optional c8y cloud profile
+        #[clap(long)]
+        profile: Option<ProfileName>,
+
+        /// Cumulocity external id of the device child on which the file has to be attached.
```

Comment:
The upload could be attached to any device/service which is owned by the main device.

```suggestion
        /// Cumulocity external id of the device/service on which the file has to be attached.
```

Reply:
Fixed by https://github.com/thin-edge/thin-edge.io/pull/3274/commits/cb54862459c13762dfb42f1b5924585a1080f019